libraries/libfc/include/fc/crypto/sha3.hpp

```diff
@@ -32,12 +32,23 @@ class sha3
 	static sha3 hash(const std::string& s, bool is_nist=true) { return hash(s.c_str(), s.size(), is_nist); }
 	static sha3 hash(const sha3& s, bool is_nist=true) { return hash(s.data(), sizeof(s._hash), is_nist); }
 
-	template <typename T>
-	static sha3 hash(const T &t, bool is_nist=true)
+	struct keccak {};
+	struct nist {};
```

Comment:
There didn't appear to be any existing usages of `sha3::hash()`. I opted to use tag dispatching to select between the two sha3 options and still have multiple arguments. The downside with this is that can't use sha3's hash() somewhere the hash type itself is templated. e.g.
```
template <typename HashType>
HashType do_something(unsigned x) {
   return HashType::hash(x);
}
```